server/cluster/cluster.go

```diff
@@ -418,62 +413,106 @@ func (c *RaftCluster) Context() context.Context {
 	return nil
 }
 
+// GetCoordinator returns the coordinator.
+func (c *RaftCluster) GetCoordinator() *coordinator {
+	return c.coordinator
+}
+
 // GetOperatorController returns the operator controller.
 func (c *RaftCluster) GetOperatorController() *schedule.OperatorController {
-	c.RLock()
-	defer c.RUnlock()
 	return c.coordinator.opController
 }
 
-// GetBasicCluster returns the basic cluster.
-func (c *RaftCluster) GetBasicCluster() *core.BasicCluster {
-	c.RLock()
-	defer c.RUnlock()
-	return c.core
-}
-
 // GetRegionScatter returns the region scatter.
 func (c *RaftCluster) GetRegionScatter() *schedule.RegionScatterer {
-	c.RLock()
-	defer c.RUnlock()
 	return c.coordinator.regionScatterer
 }
 
 // GetRegionSplitter returns the region splitter
 func (c *RaftCluster) GetRegionSplitter() *schedule.RegionSplitter {
-	c.RLock()
-	defer c.RUnlock()
 	return c.coordinator.regionSplitter
 }
 
-// GetHeartbeatStreams returns the heartbeat streams.
-func (c *RaftCluster) GetHeartbeatStreams() *hbstream.HeartbeatStreams {
-	c.RLock()
-	defer c.RUnlock()
-	return c.coordinator.hbStreams
+// GetMergeChecker returns merge checker.
+func (c *RaftCluster) GetMergeChecker() *checker.MergeChecker {
+	return c.coordinator.checkers.GetMergeChecker()
```

Comment:
Removing the `Lock` LGTM, however, should we consider the situation that `c.coordinator` is nil(IsRunning==false)?  
Here contains two issues:
1. panic with nil: if the c.coordinator is nil
2. data race: if there exists a goroutine that is starting the server(updating the c.coordinator) at the same time

If we are the only programmer who knows how to use the lib, we should run `c.Start` successfully at first. However the later programmer may not know this condition, so to be a better `lib`, Could you please check `c.IsRunning` first(which can both avoid the nil issue and keep thread safe)?